### ISSUE TITLE
cmd/compile/internal/ssa: "panic during regalloc" in `(*sparseSet).add` on linux/ppc64le

### ISSUE THREAD

**Issue by bcmills** [golang/go#52924]
`greplogs -l -e '(?ms)panic during regalloc .*cmd/compile/internal/ssa\.\(\*sparseSet\)\.add'`
[2022-05-13T21:33:25-4170084/linux-ppc64le-power9osu](https://build.golang.org/log/7ed8e0f9e7c444b00df7914ff6c7a830f2066c4e)

(attn @randall77 @dr2chase @cherrymui; CC @golang/ppc64)

**Comment by bcmills** [comment]
There have been several changes in `stackalloc.go` during the Go 1.19 cycle, so marking as release-blocker at least until we can determine whether this is a regression.

**Comment by dr2chase** [comment]
It doesn't fail using the obvious cross-compilation:
```
GOOS=linux GOARCH=ppc64le go test -c encoding/json
```
which means, something wrong in that compiler?
About to try it on a gomote....

**Comment by dr2chase** [comment]
Oh joy.  I've already verified that it doesn't reproduce (once) with
```
GOPPC64=power9 GOOS=linux GOARCH=ppc64le go test -c encoding/json
```
nor (once) on a plain ppc64le gomote.

**Comment by dr2chase** [comment]
On the failing builder, with a normally built compiler
```
for i in {1..100} ; do  go clean -cache ; go test -c encoding/json ; echo $i done ; done
```
over lunch worked fine.
Am about to retry with `GOPPC64=power9` built compiler.

**Comment by ceseo** [comment]
I can't reproduce it either (POWER9, with either `GOPPC=power9` and `GOPPC=power8`.

**Comment by cherrymui** [comment]
> There have been several changes in stackalloc.go during the Go 1.19 cycle

I don't think we have recent changes in stackalloc.go. The most recent change I saw is https://go-review.googlesource.com/c/go/+/312292 , which is a year old.

I also built the compiler in race mode on PPC64LE and couldn't find a race or a crash in a reasonable amount of time.

Given the above, and it does not occur on a first class port, I think it is not a release blocker. Thanks.

**Comment by ianlancetaylor** [comment]
As far as I can see this has happened exactly once.

Rolling forward to 1.20, but if it doesn't happen again we can close this.

**Comment by bcmills** [comment]
`greplogs -l -e '(?ms)panic during regalloc .*cmd/compile/internal/ssa' --since=2022-05-16`
[2022-08-31T21:10:41-52d9e7f/linux-ppc64-sid-buildlet](https://build.golang.org/log/f329080009702b9158c2f520dc2da09679df665b)

**Comment by bcmills** [comment]
(That time was in `(*sparseMap).set`, but it seems like a close enough fit.)

**Comment by mknyszek** [comment]
CC @golang/ppc64

**Comment by pmur** [comment]
I suspect the most recent flake here is another manifestation of #64290. The failed index is 0x80346d1, and if the high bit is cleared, the index is valid.
